### PR TITLE
Builtin method for check_provisioned

### PR DESCRIPTION
**Pull request by mkanoor**
A builtin method is run directly in the Automate Engine and doesn't have to be run via DRb, so its faster.

This is an example of invoking the check_provisioned for VM and Instance provisioning.

Currently the check_provisioned method for vm and instance provisioning is an inline method
For testing you do would have to do the following steps

1. Create a new domain, mark it as enabled
2. Copy the following 2 methods to the new domain
 a. /ManageIQ/Infrastructure/VM/Provisioning/StateMachines/Methods/check_provisioned
 b. /ManageIQ/Cloud/VM/Provisioning/StateMachines/Methods/check_provisioned
3. Copying the methods gives you the exact path of where the method should be with classes and namespaces
4. Delete the above 2 methods and add them back in as builtin
5. Make sure that the name is check_provisioned, builtin method names match to the internal name.

This method takes an optional single input parameter 
 

- ae_retry_interval                     Time in seconds to sleep between retries default is 60 seconds

### CODE REVIEW

File: lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_builtin_method/miq_ae_task_builtin.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      extend ActiveSupport::Concern
+
+      module ClassMethods
+        def miq_check_provisioned(obj, inputs = {})
+          root = obj.workspace.root
+          task = root['miq_provision']
+          raise ArgumentError, "miq_provision not specified" unless task
+
+          result = task.statemachine_task_status
+          $miq_ae_logger.info("Builtin ProvisionCheck returned <#{result}> for state <#{task.state}> and status <#{task['status']}>")
```
Comment (tinaafitz) [inline lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_builtin_method/miq_ae_task_builtin.rb]: @mkanoor Can we replace "ProvisionCheck" here with "check_provisioned"?

File: lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_builtin_method/miq_ae_task_builtin.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+          set_ae_result(result, task, root, inputs)
+        end
+
+        def set_ae_result(result, task, root, inputs)
+          case result
+          when 'error'
+            root['ae_result'] = 'error'
+            reason = task.message
+            $miq_ae_logger.error("ProvisionCheck error <#{reason}>")
```
Comment (tinaafitz) [inline lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_builtin_method/miq_ae_task_builtin.rb]: @mkanoor Can we change it here as well?

File: lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_builtin_method/miq_ae_task_builtin.rb (hunk 1 above)
Comment (lfu) [reply lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_builtin_method/miq_ae_task_builtin.rb]: @mkanoor Are we going to have one file for all task built in methods?

Comment (mkanoor) [reply lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_builtin_method/miq_ae_task_builtin.rb]: @lfu I think we need 1 file for common functions like tasks (check_provisioned, check_service_provisioned, update_provision_status) They are all reading/updating the MiqRequestTasks

**Pull request closed without merging**